clusterman/kubernetes/util.py

```diff
@@ -60,7 +60,18 @@ def __init__(self, kubeconfig_path: str, client_class: Type) -> None:
         :param Type client_class: k8s client class to initialize
         """
         try:
-            kubernetes.config.load_kube_config(kubeconfig_path, context=os.getenv("KUBECONTEXT"))
+            """
+            https://kubernetes.io/docs/concepts/containers/container-environment/#container-environment
+            Every pod in k8s get some default enviornment variable injected, Including KUBERNETES_SERVICE_HOST
+            which points to K8s service. We are using this variable to distinguise between when cluterman is started in
+            a pod vs when it's started on host. For clusterman services running inside a k8s cluster, We priiotise using
+            K8s Service account since that let us avoid creating any kubeconfig in advance.
+            For clusterman CLI invokation we continue using provided KUBECONFIG file
+            """
+            if os.getenv("KUBERNETES_SERVICE_HOST"):
```

Comment:
I think what we (well, other Yelpers) have been doing is checking if `/var/run/secrets/kubernetes.io/serviceaccount` exists and taking that to mean that we're running inside k8s

that said, not a blocker if you prefer this method instead

Reply:
https://github.com/Yelp/paasta/blob/master/paasta_tools/cli/cmds/local_run.py#L659
seems like we only overwrite variables for MESOS.